smart_contracts.rst

```diff
@@ -337,6 +461,35 @@ Unlocks all pending transfers by providing the entire merkle tree of pending tra
     ``unlock`` must be called after ``settleChannel`` because it needs the ``locksroot`` from the latest :term:`balance proof` in order to guarantee that all locks have either been unlocked or have expired.
 
 
+.. _settle-computation:
+
+Computation of balances
+~~~~~~~~~~~~~~~~~~~~~~~
+
+There is an elaborate algorithm to safely compute the balances, avoiding simple errors like over- and underflows as well as vulnerabilities for cheating.
+A possible attempt at cheating would be to present an outdated balance proof before an unlocking event, so that the locked amount would be assessed too high.
+
+The **net deposit** of a participant (``ND1, ND2``) is their deposited amount minus their withdrawn amount.
+
+The sum of net deposits is the **total available deposit**: ``TAD = ND1 + ND2``.
+It is computed from the channel state saved in the contract, with all balance proofs that have been presented so far.
+In any valid settlement the following invariants hold:
+
+- The ``TAD`` will be nonnegative.
+- The sum of locked and transferred amounts of both participants will be equal to the ``TAD``.
+
+The **maximum possible receivable amount** of participant 1 is computed as ``MR1 = ND1 + T1 + L1 - T2 - L2``, with ``T1, T2, L1, L2`` the transferred and locked amounts passed to ``settleChannel``.
+
+This value, capped by the total available deposit, is assigned to participant 1 as their balance: ``B1 = min(TAD, MR1)``
+
+The balance of participant 2 is then deducted by the invariant stated above: ``B2 = TAD - B1``
+
+The **actually locked amounts** will be the locked amounts capped by the respective partners' balance: ``La1 = min(L1, B2); La2 = min(L2, B1)``
+
+Finally, the **actually transferred amounts** will be balances minus actually locked amounts: ``Ta1 = B1 - La2; Ta2 = B2 - La1``
+
+If ``Ta1 > 0`` or ``Ta2 > 0``, they are transferred in the settlement of the channel.
```

Comment:
As said above, I would concentrate on the constraints here, rather than the algorithm. The specs should be guidelines for anyone wanting to implement the protocol. The settlement algorithm could possibly be implemented in other ways too.
I would rather have this in another PR. And I will actually make one with what I wrote in https://github.com/raiden-network/raiden-contracts/issues/188#issuecomment-404752095 if you don't mind.

Reply:
I did the PR here: https://github.com/raiden-network/spec/pull/99